File: rust/src/models/create_subscription_add_on.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::{
    add_on_id::AddOnId, subscription_add_on_customization::SubscriptionAddOnCustomization,
};

#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct CreateSubscriptionAddOn {
    pub add_on_id: AddOnId,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub customization: Option<SubscriptionAddOnCustomization>,
}

impl CreateSubscriptionAddOn {
    pub fn new(add_on_id: AddOnId) -> Self {
        Self {
            add_on_id,
            customization: None,
        }
    }
}

File: typescript/src/models/createSubscriptionAddOn.ts
// this file is @generated
import { type AddOnId, AddOnIdSerializer } from "./addOnId";
import {
  type SubscriptionAddOnCustomization,
  SubscriptionAddOnCustomizationSerializer,
} from "./subscriptionAddOnCustomization";

export interface CreateSubscriptionAddOn {
  addOnId: AddOnId;

  customization?: SubscriptionAddOnCustomization | null;

  quantity?: number;
}

export const CreateSubscriptionAddOnSerializer = {
  _fromJsonObject(object: any): CreateSubscriptionAddOn {
    return {
      addOnId: AddOnIdSerializer._fromJsonObject(object["add_on_id"]),
      customization:
        object["customization"] != null
          ? SubscriptionAddOnCustomizationSerializer._fromJsonObject(
              object["customization"]
            )
          : undefined,
      quantity: object["quantity"],
    };
  },

  _toJsonObject(self: CreateSubscriptionAddOn): any {
    return {
      add_on_id: AddOnIdSerializer._toJsonObject(self.addOnId),
      customization:
        self.customization != null
          ? SubscriptionAddOnCustomizationSerializer._toJsonObject(self.customization)
          : undefined,
      quantity: self.quantity,
    };
  },
};
